### DISCUSSION TITLE
About using the word "icon" in the <title> tag of files

### DISCUSSION THREAD

**Discussion by gizmecano** [simple-icons/simple-icons#5186] [Ideas]
I would like to point out a small questioning which is not a real issue by itself. It's about the use of the word "_icon_" in the `<title>` of `SVG` files.

Indeed, when working on a project that doesn't use English as its primary language (whether it is a site in another language or a multilingual site), all the icons might display a tooltip that turns out to be written in a language which isn't the same as the one mainly used in the site. IMHO, this seems to potentially create some confusion, especially for people who absolutely don't understand English.

Then, even if the exact way in which these tooltips are understood and processed by screen readers, using a specific language (_i.e._ the one used by the project) in this kind of tag could maybe become problematic in terms of accessibility. In various languages, the term spelled "_icon_" can be pronounced in a totally different way and may come closer to other terms of distinct meaning (thus in French, the pronunciation of the word "_icon_" could sounds like another word  which is an offensive word).

Finally, even if currently only one file contains this exact four letters string (cf. #2158 and #2165), the possible lack of clarity of a title where these letters would be repeated several times in a successive manner seems difficult to deny (and particularly again in the potential case of non-visual browsers).

Of course, `SVG` files are of visual interest first, but many people use [various ways to make graphics files accessible](https://css-tricks.com/accessible-svgs/) (_e.g._ think about `alt` tags on bitmap images).

Accordingly, I presume that it would probably be relevant if the `<title>` included in the `SVG` files were just the same as the titles of the files present in the directory or saved as a value in the `JSON` file.

**Comment by mondeja** [comment]
Personally, I agree, I've never understood why the repeat of "icon" word in all the files. That said, keep in mind that you can easily remove the `title` tag or change their content by another text while preprocessing the SVGs, just as if you wanted to add color to them.

**Reply by gizmecano** [reply]
> (...) keep in mind that you can easily remove the `title` tag or change their content by another text (...)

Of course, you're absolutely right, and that's what I just did for a project I'm working on.

**Comment by PeterShaggyNoble** [comment]
Thanks for the feedback, @gizmecano.

We do have plans, as a follow-up to #4797, to somehow add support for internationalised icons and this is something that could be addressed as a part of that. Of course, it would require us to know the word for "icon" in each language and to store then somewhere in order to construct the new `<title>`. Between the team of maintainers we do have a fair few languages covered already but, obviously, not all so the addition of the translations would need to be an ongoing process with a fallback to the English if we don't yet have a translation in a particular language.

Of course, we'd also need to address the fact that the _structure_ of the `<title>` would need to be different in some languages (e.g., "Icono de [BRAND]" in Spanish) but we can figure that out when the time comes. The other issue is that, in order to provide an icon with the word "icon" translated, the icon being used would need an alias in the language the user wants to use which won't always be the case so we may need to consider another solution.

**Reply by gizmecano** [reply]
> We do have plans (...) to somehow add support for internationalised icons and this is something that could be addressed as a part of that. 

This kind of effort is appreciated and should be commended.
 
> (...) we'd also need to address the fact that the _structure_ of the `<title>` would need to be different in some languages (...)

Nevertheless, in the idea of keeping the use of all the icons as easy and simple as possible (as the main title of the project presents it), simply remove this term seems more convenient.

Of course, such a potential change does not prevent in any way working on i18n / L10N of the other parts of the project, although it seems like a colossal task.

😉

**Reply by ericcornelissen** [reply]
I don't think we should attempt to provide localised versions of  "[Brand name] icon", that would lead to an explosion of the size of the package that seems mostly unnecessary to me. Unless we choose to only generate those on demand, but that ignores CDN users that use the SVGs directly.

**Comment by ericcornelissen** [comment]
I think the word "icon" is included because it is **arguably** more descriptive than not including it (the "ICON icon" example seems like a problem with the brand name to me), but I can definitely see that omitting it has advantages the primary one being indeed that it translates better. Additionally it makes the SVGs and JavaScript files smaller and in some use cases just the brand name may be more accessible. So yes, I would be fine if we remove it but I think, just to be safe, we should do that with a major release.


> Of course, SVG files are of visual interest first, but many people use various ways to make graphics files accessible (e.g. think about alt tags on bitmap images).

This is a bit of a weak argument since, as far as I can tell, all the alternatives way of describing the image override or add to the `<title>` value we provide, but that's besides the point :slightly_smiling_face:

**Reply by gizmecano** [reply]
I admit that the argument on accessibility can be discussed depending on the case... 😉

Nonetheless, the benefits you have discerned in terms of i18n/L10N and about the _weight_ of the project would likely be profitable.

And, of course, it seems consistent that such a change can only be considered as a _major_ change.

**Comment by fbernhart** [comment]
I'm as well in favour of dropping "icon" in the title tag.

Maybe we can get this idea merged before the next major release? (which is scheduled for the end of May)

**Reply by ericcornelissen** [reply]
As there seems to be a consensus amongst the maintainers, I think so yes.

**Comment by ericcornelissen** [comment]
I opened https://github.com/simple-icons/simple-icons/issues/5478 so that it can be added to [the v5 milestone](https://github.com/simple-icons/simple-icons/milestone/4).

I will lock this conversation so that any further discussion regarding this topic happens in the issue.
